### ISSUE TITLE
/doc/reference/glossary/

### ISSUE THREAD

**Issue by behoppe** [OpenCilk/www.opencilk.org#16]
A glossary would be a great asset on the site. For some concepts there are multiple terms from different teams or different eras (e.g. span, depth, critical path length). We'll help people get that straight. Charles provided me with the Parallelism chapter from his book to make a starting list of entries.

I wonder if it would be possible to have Eleventy parse each site page so that when these words appear (perhaps with some markup like &span&) then they are rendered as a link to the glossary entry.

**Comment by behoppe** [comment]
Started glossary based on \defn{} terms in the Parallel chapter of the Algorithms book. Each term gets a file in the /src/doc/reference/glossary folder. I have entered a few definitions and set up an Eleventy collection that you can edit with Netlify CMS.

One wrinkle in the current design is that each entry holds its definition in its front matter. For the sake of editing, I would prefer the definition to go in the body/content of the entry, but I am not sure how to pull that body/content into a unified glossary page. (Whereas front matter is easy to pull into a unified glossary page.)

**Comment by behoppe** [comment]
@cleiserson I was unable to resolve Netlify CMS issues (instead calling Netlify CMS into question per #100); however, I did make some important updates to the glossary setup.

- I figured out how to programmatically render the "body" of each entry as a list on a single page, and so the clunky use of front matter to hold a "defn" field is over. I have reconfigured every glossary entry to have a "title" in its front matter, followed by its definition in the body of the post. And Netlify CMS is reconfigured to edit the content in that format.
- I also implemented a `{% defn "term", "optional text" %}` shortcode, which I will document tomorrow. (Short version: It does exactly what we have discussed last week.)
